hive_setup.sh

```diff
@@ -0,0 +1,278 @@
+#!/usr/bin/env bash
+set -xe
+
+source logging.sh
+source common.sh
+source utils.sh
+source rhcos.sh
+source ocp_install_env.sh
+
+function generate_hive_manifest() {
+    cat > "ocp/${CLUSTER_NAME}/manifests.yaml" <<EOF
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${CLUSTER_NAME}-pull-secret
+type: kubernetes.io/dockerconfigjson
+stringData:
+  .dockerconfigjson: |-
+    $(echo $PULL_SECRET | jq -c .)
+
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: ${CLUSTER_NAME}-ssh-private-key
+stringData:
+  ssh-privatekey: |-
+$(cat ~/.ssh/id_rsa | sed 's/^/    /g')
+
+---
+apiVersion: hive.openshift.io/v1
+kind: ClusterDeployment
+metadata:
+  name: ${CLUSTER_NAME}
+  annotations:
+    # do not retry if first deploy fails
+    hive.openshift.io/try-install-once: "true"
+spec:
+  baseDomain: ${BASE_DOMAIN}
+  clusterName: ${CLUSTER_NAME}
+  controlPlaneConfig:
+    servingCertificates: {}
+  platform:
+    bareMetal:
+      libvirtSSHPrivateKeySecretRef:
+        name: ${CLUSTER_NAME}-ssh-private-key
+  provisioning:
+    installConfigSecretRef:
+      name: ${CLUSTER_NAME}-install-config
+    sshPrivateKeySecretRef:
+      name: ${CLUSTER_NAME}-ssh-private-key
+#    manifestsConfigMapRef:
+#      name: ${CLUSTER_NAME}-install-manifests
+    releaseImage: "${OPENSHIFT_RELEASE_IMAGE}"
+    sshKnownHosts:
+$(ssh-keyscan -H ${PROVISIONING_HOST_IP} 2>/dev/null | sed -e 's/^/      - "/g' -e 's/$/"/g')
```

Comment:
I can't read this but if it's coming out as '      - "SSHKEY"' we're good. :)

Reply:
On my host ssh-keyscan gives me

```
# localhost:22 SSH-2.0-OpenSSH_7.4
|1|LxI0OGI1Bcr97HslbGsrmWamUcM=|SnWlTu5a6+Fc+ZKtrw8VbKmqQjs= ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQCh5D3FqClxnEZGZiZD9fclaVjkzZ98UX+xSBt+v4Q3ashpVhtYda61UTUmbXh9u8J44bYO5DS+OKtbDiBqQSB6p6VuxcIfjvapasUt7awAZIwBMUs2nYA9YA7F+0qpl14M+iK0zdbTqXhDn/KWfGxMK7YhZwPdmHNXV8V3B7yqmpc1skm3B4F5jgMWtY+a80e609SCtyVUaiRwibNILEvFM4YEMng9W9VAhweo1c4wt/hcwbLLQdu+XLrr+56WOy+9Ns9dfJBV4cHyV6Oflb1c2jcG/+FE+1jE1qpADP/lMaX9IPyil8sfPpBRWmpmy2La5pX8Wpc3zfMiz1VcYLAJ
# localhost:22 SSH-2.0-OpenSSH_7.4
|1|ZEpP9jvoFXWI9tbrTvrAlDXGlVE=|1CdDt0RJSixKsX6hRfbI4F1FVHE= ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBKWjJRzeUVuZs4yxSy4eu45xiANFIIbwE3e1aPzGD58x/NX7Yf+S8eFKq4RrsfSaK2hVJyJjvVIhUsU9z2sBJP8=
# localhost:22 SSH-2.0-OpenSSH_7.4
|1|rPeaRJpIzfn/i/ySG7ZHN3NdUjM=|xdcxdIXslQ/NQl7bIyJmToQ6gas= ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIByDGt93ltGsuJWupi40PEn2VtiYuWw0/Bx2d9PYN1pQ
```

The comment lines go to stderr, so those are dropped.

The sed call adds some spaces and `- "` to the start of each line and `"` to the end of each line.

That gives us all of the host keys for the host in a YAML list.